File: Bit Manipulation/XOR-Range-Queries.cpp
vector<int> solve(vector<int>& nums, vector<vector<int>>& queries) {
  std::vector<int> prefix_xors = {0};
  prefix_xors.reserve(nums.size() + 1);
  for (int num : nums) {
    prefix_xors.push_back(prefix_xors.back() ^ num);
  }

  std::vector<int> answers;
  answers.reserve(queries.size());
  for (const auto& query : queries) {
    answers.push_back(prefix_xors[query[1] + 1] ^ prefix_xors[query[0]]);
  }
  return answers;
}

File: Bit Manipulation/XOR-Range-Queries.py
class Solution:
  def solve(self, nums, queries):
    prefix_xors = [0];
    for num in nums:
      prefix_xors.append(prefix_xors[-1] ^ num)

    return [prefix_xors[query[1] + 1] ^ prefix_xors[query[0]] for query in queries]
